### PR TITLE
Spain Calendar

### PR DESCRIPTION
**Pull request by dmichalowicz**
Exchange calendar for the Bolsa de Madrid.

### CODE REVIEW

File: tests/test_xmad_calendar.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        # In general, holidays falling on a weekend should not be made up
+        # during the week.
+        expected_sessions = [
+            # In 2010, Labour Day fell on a Saturday, so the market should be
+            # open on both the prior Friday and the following Monday.
+            pd.Timestamp('2010-04-30', tz='UTC'),
+            pd.Timestamp('2010-05-03', tz='UTC'),
+            # Christmas also fell on a Saturday, meaning Boxing Day fell on a
+            # Sunday. The market should still be open on both the prior Friday
+            # and the following Monday.
```
Comment (yankees714) [inline tests/test_xmad_calendar.py]: 😣

File: tests/test_xmad_calendar.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        expected_holidays = [
+            pd.Timestamp('2006-01-06', tz='UTC'),  # Epiphany
+            pd.Timestamp('2003-08-15', tz='UTC'),  # Assumption Day
+            pd.Timestamp('2004-10-12', tz='UTC'),  # National Day
+            pd.Timestamp('2004-11-01', tz='UTC'),  # All Saints Day
+            pd.Timestamp('2004-12-06', tz='UTC'),  # Constitution Day
+            pd.Timestamp('2004-12-08', tz='UTC'),  # Immaculate Conception
+            pd.Timestamp('2004-12-24', tz='UTC'),  # Christmas Eve
+        ]
```
Comment (yankees714) [inline tests/test_xmad_calendar.py]: Should we include New Year's Eve here?

Comment (dmichalowicz) [reply tests/test_xmad_calendar.py]: Yup good catch.

File: trading_calendars/exchange_calendar_xmad.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    Close Time: 5:30 PM, CET (Central European Time)
+
+    Regularly-Observed Holidays:
+      - New Year's Day
+      - Good Friday
+      - Easter Monday
+      - Labour Day
+      - Christmas Day
+      - Boxing Day
+
```
[the remark this answers is not in the document]
Comment (dmichalowicz) [reply trading_calendars/exchange_calendar_xmad.py]: Ya I can add that.

**Pull request merged**